project/ServerlessDeployPlugin.scala

```diff
@@ -18,19 +32,45 @@ object ServerlessDeployPlugin extends AutoPlugin {
   override def requires: Plugins = UniversalPlugin
 
   override lazy val projectSettings = Seq(
-    serverlessDeployCommand := "serverless deploy --verbose".split(' ').toSeq,
+    serverlessPackageCommand := "serverless package --verbose".split(' ').toSeq,
+    getStackNameCommand := s"serverless print --path provider.stackName --format text".split(' ').toSeq,
+    cloudformationDeployCommand := "aws cloudformation deploy --capabilities CAPABILITY_NAMED_IAM".split(' ').toSeq,
+
     deploy := Def.inputTask {
-      import scala.sys.process._
+      val stage = Stage.parser.parsed.name.toLowerCase
+      val artifactPath = (Universal / packageBin).value.toString
 
-      val baseCommand = serverlessDeployCommand.value
-      val exitCode = Process(
-        baseCommand ++ Seq("--stage", Stage.parser.parsed.name.toLowerCase),
+      val serverlessExitCode = Process(
+        serverlessPackageCommand.value ++ Seq("--stage", stage),
         Option((ThisBuild / baseDirectory).value),
-        "ARTIFACT_PATH" -> (Universal / packageBin).value.toString,
+        "ARTIFACT_PATH" -> artifactPath,
       ).!
 
-      if (exitCode == 0) exitCode
-      else throw new IllegalStateException("Serverless returned a non-zero exit code. Please check the logs for more information.")
+      if (serverlessExitCode != 0) throw new IllegalStateException("Serverless returned a non-zero exit code. Please " +
+        "check the logs for more information.")
+
+      val stackName = Process(
+        getStackNameCommand.value ++ Seq("--stage", stage),
+        Option((ThisBuild / baseDirectory).value),
+        "ARTIFACT_PATH" -> artifactPath,
+      ).!!.trim
+
+      val cloudformationTemplate = (ThisBuild / baseDirectory).value /
```

Comment:
We might want to separate out this step that modifies the Cloudformation template into a separate sbt task, which would allow the user to examine the output of the modification before it's submitted to Cloudformation, but it's not a requirement and we could always do it as a followup if we want.

(In other words, there could be an sbt task to modify the generated template (that would return the path to the modified template), and then a task to do the artifact upload and deploy that depends on the modification task.)

Reply:
Updated.